Solar-System.rb

```diff
@@ -0,0 +1,225 @@
+        distance_info = ""
+        @planets.each do |other_planet|
+          distance_info += "The distance from #{planet_name} to #{other_planet.name} is #{(each_planet.distance_from_the_sun.split[0].to_i - other_planet.distance_from_the_sun.split[0].to_i).abs} miles.\n"
+        end
+        return distance_info
+      end
+    end
+  end
+
+end
+# Create a new class Planet
+class Planet
+  attr_reader :name, :diameter, :gravity, :year_length, :distance_from_the_sun
+
+  # Creat a constructor to initialize the class
+  def initialize(name, diameter, gravity, year_length, distance_from_the_sun)
+    @name = name
+    @diameter = diameter
+    @gravity = gravity
+    @year_length = year_length
+    @distance_from_the_sun = distance_from_the_sun
+  end
+
+  def show_attributes
+    plant_attributes += "#{@name}'s attributes:"
+    plant_attributes += "\nIt has a diameter of #{@diameter}."
+    plant_attributes += "\nIts gravity is #{@gravity}."
+    plant_attributes += "\nIt takes #{@year_length} to spin on Axis."
+    plant_attributes += "\nIts distance from the sun is #{@distance_from_the_sun}.\n"
+    return plant_attributes
+  end
+
+end
+
+# The planet pool is an array holding hashes of planet info.
+pool = [
+  {
+    name: "Mercury",
+    diameter: "3,301 miles",
+    gravity: 0.36,
+    year_length: "59 days",
+    distance_from_the_sun: "36000000 miles",
+  },
+  {
+    name: "Venus",
+    diameter: "7,521 miles",
+    gravity: 0.9,
+    year_length: "243 days",
+    distance_from_the_sun: "67000000 miles",
+  },
+  {
+    name: "Earth",
+    diameter: "12,765 miles",
+    gravity: 1,
+    year_length: "23 hours, 56 mins",
+    distance_from_the_sun: "92960000 miles",
+  },
+  {
+    name: "Jupiter",
+    diameter: "88,846 miles",
+    gravity: 2.64,
+    year_length: "9 hours, 55 mins",
+    distance_from_the_sun: "483500000 miles",
+  },
+  {
+    name: "Saturn",
+    diameter: "74,900 miles",
+    gravity: 1.16,
+    year_length: "10 hours, 39 mins",
+    distance_from_the_sun: "888750000 miles",
+  },
+  {
+    name: "Uranus",
+    diameter: "31,763 miles",
+    gravity: 1.11,
+    year_length: "17 hours, 14 mins",
+    distance_from_the_sun: "1783744000 miles",
+  },
+  {
+    name: "Neptune",
+    diameter: "30,779 miles",
+    gravity: 1.21,
+    year_length: "16 hours, 7 mins",
+    distance_from_the_sun: "2797770000 miles",
+  },
+  {
+    name: "Mars",
+    diameter: "4,222 miles",
+    gravity: 0.38,
+    year_length: "24 hours, 37 mins",
+    distance_from_the_sun: "141700000 miles",
+  }
+]
+
+# Create a method to create a pool of Planet instance by using pool information which is an array
+def pool_to_planet(pool)
+  # Create a variable to store all new Planet s
+  library_of_planets = []
+  pool.each do |planet_info|
+    library_of_planets << planet = Planet.new(planet_info[:name], planet_info[:diameter], planet_info[:gravity],
+      planet_info[:year_length], planet_info[:distance_from_the_sun])
+    end
```

Comment:
Watch indentation through here - it looks like Atom got confused, and the rest of your file ended up getting indented 2 extra spaces!

Reply:
Thanks for catching that. Incorrect indention is because I start a new line manually. Atom was confused. Changed.